install.sh

```diff
@@ -1,9 +1,103 @@
+#!/bin/bash
 
+URL=https://github.com/LiqoTech/liqo.git
+FOLDER=liqo
+HELM_VERSION=v3.2.1
+HELM_URL=https://get.helm.sh/helm-${HELM_VERSION}-linux-amd64.tar.gz
 
+function cleanup()
+{
+   export PATH=$OLD_PATH
+}
 
+# Define a string variable with a value
+commands="curl kubectl"
 
-curl -fsSL -o get_helm.sh https://raw.githubusercontent.com/helm/helm/master/scripts/get-helm-3
-chmod 700 get_helm.sh
-./get_helm.sh
+
+trap cleanup EXIT
+echo "[PRE-INSTALL]: Checking all pre-requisites are met"
+# Iterate the string variable using for loop
+for val in $commands; do
+    if command -v $val > /dev/null; then
+      echo "[PRE-INSTALL]: $val correctly installed"
+    else
+      echo "[PRE-INSTALL] [FATAL] : $val not found. Exiting"
+      exit 1
+    fi
+done
+
+set -e
+
+mkdir -p /tmp/liqo
+cd /tmp/liqo || exit 1
+mkdir -p /tmp/liqo
+cd /tmp/liqo || exit 1
+mkdir -p bin/
+cp linux-amd64/helm bin/
+export OLD_PATH=$PATH
+export PATH=$PATH:/tmp/liqo/bin/
+echo " [PRE-INSTALL] [HELM] Checking HELM installation..."
+HELM_INSTALLATION=false
+  if command -v helm > /dev/null; then
+    HELM_INSTALLED_VERSION=$(helm version --template "{{ .Version }}")
+    echo "[PRE-INSTALL]: Helm, found version: $HELM_INSTALLED_VERSION"
+    if [ "$HELM_INSTALLED_VERSION" != "$HELM_VERSION" ]; then
+         HELM_INSTALLATION=true
+         echo " [PRE-INSTALL] [HELM] found but having a potential incompatible version... Installing it"
+    fi
+  else
+    echo " [PRE-INSTALL] [HELM]: Helm not found... Installing it"
+    HELM_INSTALLATION=true
+fi
+if [ $HELM_INSTALLATION = true ]; then
+    echo " [PRE-INSTALL] [HELM]: Installing Helm $HELM_VERSION"
+    curl --fail -LO $HELM_URL > /dev/null
+    tar zxf helm-v3.2.3-linux-amd64.tar.gz
+fi
+
+if [ ! -d "$FOLDER" ] ; then
+    git clone "$URL" 
+fi
+
+echo "[PRE-INSTALL]: Collecting installation variables. The installer will retrieve installation parameters from your
+ Kubernetes cluster. Fill free to override them, by launching it with those environment variables set in advance."
+if [ -z "$POD_CIDR" ]
+then
+  echo "[PRE-INSTALL]: Kubeconfig variable is set to: $KUBECONFIG"
+else
+  echo "[PRE-INSTALL]: Kubeconfig variable is set to: ~/.kube/config"
+fi
+if [ -z "$POD_CIDR" ]
+then
+    POD_CIDR=$(kubectl cluster-info dump | grep -m 1 cluster-cidr | sed 's/--cluster-cidr=//' | sed 's/[,"]//g' | sed 's/ //g')
```

Comment:
1. Maybe here you could directly filter the output in advance to avoid downloading too much data (jsonpath or gotemplate?)
2. You could replace the multiple `sed` with a single `grep` with regex and capture group